website/project/views/drafts.py

```diff
@@ -0,0 +1,282 @@
+    if not schema_name:
+        raise HTTPError(http.BAD_REQUEST)
+
+    schema_version = data.get('schema_version', 1)
+    schema_data = data.get('schema_data', {})
+
+    meta_schema = get_schema_or_fail(
+        Q('name', 'eq', schema_name) &
+        Q('schema_version', 'eq', schema_version)
+    )
+    draft = node.create_draft_registration(
+        user=auth.user,
+        schema=meta_schema,
+        data=schema_data,
+        save=True,
+    )
+    return serialize_draft_registration(draft, auth), http.CREATED
+
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def new_draft_registration(auth, node, *args, **kwargs):
+    """Create a new draft registration for the node
+
+    :return: Redirect to the new draft's edit page
+    :rtype: flask.redirect
+    :raises: HTTPError
+    """
+
+    data = request.values
+
+    schema_name = data.get('schema_name')
+    if not schema_name:
+        raise HTTPError(
+            http.BAD_REQUEST,
+            data={
+                'message_short': 'Must specify a schema_name',
+                'message_long': 'Please specify a schema_name'
+            }
+        )
+
+    schema_version = data.get('schema_version', 1)
+
+    # can return 404 if the schema does not exist
+    # can return 400 if the schema is in the DB twice (should never happen)
+    meta_schema = get_schema_or_fail(
+        Q('name', 'eq', schema_name) &
+        Q('schema_version', 'eq', int(schema_version))
+    )
+    draft = node.create_draft_registration(
+        user=auth.user,
+        schema=meta_schema,
+        data={},
+        save=True,
+    )
+    return redirect(node.web_url_for('edit_draft_registration_page', draft_id=draft._id))
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def edit_draft_registration_page(auth, node, draft, **kwargs):
+    """Draft registration editor
+
+    :return: serialized DraftRegistration
+    :rtype: dict
+    """
+    ret = project_utils.serialize_node(node, auth, primary=True)
+    ret['draft'] = serialize_draft_registration(draft, auth)
+    return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def update_draft_registration(auth, node, draft, *args, **kwargs):
+    """Update an existing draft registration
+
+    :return: serialized draft registration
+    :rtype: dict
+    :raises: HTTPError
+    """
+    data = request.get_json()
+
+    schema_data = data.get('schema_data', {})
+
+    schema_name = data.get('schema_name')
+    schema_version = data.get('schema_version', 1)
+    if schema_name:
+        meta_schema = get_schema_or_fail(
+            Q('name', 'eq', schema_name) &
+            Q('schema_version', 'eq', schema_version)
+        )
+        existing_schema = draft.registration_schema
+        if (existing_schema.name, existing_schema.schema_version) != (meta_schema.name, meta_schema.schema_version):
+            draft.registration_schema = meta_schema
+
+    draft.update_metadata(schema_data)
```

Comment:
Does taking directly from the input JSON payload have the side effect of placing arbitrary user inputs directly into the database? Our current standard is sanitize-on-entry.

(This is either the third or fourth time I've pointed it out on this view, so please let me know if there's a particular design reason for this behavior)